### PR TITLE
Update design documentation

### PR DESCRIPTION
**Pull request by slagle**
The design doc is updated to reflect the current CRD design and for
providing a high level overview and introduction of dataplane-operator.

Signed-off-by: James Slagle <jslagle@redhat.com>

### CODE REVIEW

**Review by fao89 — COMMENTED** [review]
+2

File: docs/assemblies/design.adoc [hunk 1]
```diff
@@ -1,55 +1,83 @@
...
+
+Similarities within a `OpenStackDataPlaneNodeSet` are defined by the user, and
+could be of a small scope (ansible port), or a large scope (same network
+config, nova config, provisioning config, etc). The properties that all nodes
+in a `OpenStackDataPlaneNodeSet` share is set in the `nodeTemplate` field of
+the `OpenStackDataPlaneNodeSet` spec. Node specific parameters are then defined
+under the `nodeTemplate.nodes` section specific to that node. Options defined
+here will override the inherited values from the `NodeSet`.
+
+Divding and assigning nodes to different `OpenStackDataPlaneNodeSets` is a
```
Comment (igallagh-redhat) [inline docs/assemblies/design.adoc:48]:
```suggestion
Dividing and assigning nodes to different `OpenStackDataPlaneNodeSets` is a
```

File: docs/assemblies/design.adoc [hunk 2]
```diff
@@ -1,55 +1,83 @@
...
+https://github.com/openstack-k8s-operators/openstack-ansibleee-operator[openstack-ansibleee-operator]
+to execute Ansible to deploy, configure, and orchestrate software on the nodes.
+The software is typically RPM or container based using the `podman` container
+runtime.
+
+External Data Plane Management (EDPM) is the concept of using Ansible in this
+manner to configure software on data plane nodes. Ansible is used instead of
+using native Kubernetes Workload API's (Deployment, Job, Pod, etc) and kubelet.
+While the Ansible executions themselves run on the Kubernetes cluster as native
+Kubernetes workloads, they communicate using SSH with data plane nodes and uses
```
Comment (jpodivin) [inline docs/assemblies/design.adoc:22]: use

File: docs/assemblies/design.adoc [hunk 3]
```diff
@@ -1,55 +1,83 @@
...
+The `OpenStackDataPlaneNodeSet` CRD is used to describe a logical grouping of
+nodes of a similar type. A node can only be defined in one NodeSet. This is
+analogous to the concept of "roles" in TripleO. An OpenStack data plane is
+likely to consist of multiple `OpenStackDataPlaneNodeSet` resources to describe
+groups of nodes that are configured differently.
+
+Similarities within a `OpenStackDataPlaneNodeSet` are defined by the user, and
+could be of a small scope (ansible port), or a large scope (same network
+config, nova config, provisioning config, etc). The properties that all nodes
+in a `OpenStackDataPlaneNodeSet` share is set in the `nodeTemplate` field of
```
Comment (jpodivin) [inline docs/assemblies/design.adoc:43]: are

**Review by jpodivin — APPROVED** [review]
minor nits but otherwise lgtm

File: docs/assemblies/design.adoc (hunk 2 above)
Comment (slagle) [reply docs/assemblies/design.adoc]: Done.

File: docs/assemblies/design.adoc (hunk 3 above)
Comment (slagle) [reply docs/assemblies/design.adoc]: Done.

**Pull request merged**